src/symbols.c

```diff
@@ -1686,8 +1704,16 @@ static gboolean goto_tag(const gchar *name, gboolean definition)
 }
 
 
-gboolean symbols_goto_tag(const gchar *name, gboolean definition)
+gboolean symbols_goto_tag(const gchar *name, gint pos, gboolean definition)
 {
+	GeanyDocument *doc = document_get_current();
+
+	if (plugin_extension_goto_provided(doc))
+	{
+		plugin_extension_goto_perform(doc, pos, definition);
+		return TRUE;
```

Comment:
Ad (1): `plugin_extension_goto_perform()` could be modified to return gboolean so Geany could use it if it uses the PluginExtensions interface. The LSP plugin though will always have to return TRUE here.

Reply:
1. I think it's probably fine to always return TRUE and pretend it's handled.  I'd have to check, but I doubt it'd change anything.
2. I'm not really thinking it'd be better to pass the word, as said it probably makes sense for the extension to be trusted to know what a word is for a language it handles (although you're basically saying Geany is doing better than what your LSP plugin is currently doing, but that could change).  My concern is that we are checking whether there's a word *outside* the call, meaning the extension still depends on what Geany things a word is.
   If we're happy with ignoring point 1 above, we could have this API return TRUE if the extension handled the request (which could just be "there's something that looks like a legit target, I'll try to go there", not necessarily meaning it did perform the action synchronously).

For point 2, it could lead to something like this:
```c
if (! plugin_extension_goto_perform(…))
    keybindings_send_command(GEANY_KEY_GROUP_GOTO, GEANY_KEYS_GOTO_MATCHINGBRACE);
```
or
```c
if (! plugin_extension_goto_perform(…))
    utils_beep();
```

Reply:
> Ad (1): `plugin_extension_goto_perform()` could be modified to return gboolean so Geany could use it if it uses the PluginExtensions interface. The LSP plugin though will always have to return TRUE here.

Well, the LSP plugin would look for the word and return FALSE if it didn't find any.

Reply:
OK, sounds good.